apps/client/src/features/app-settings/panel/project-panel/RenameProjectForm.tsx

```diff
@@ -0,0 +1,65 @@
+import { FormControl, Input, Button } from '@chakra-ui/react';
+import { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+
+import style from './ProjectPanel.module.scss';
+
+export type RenameProjectFormValues = {
+  filename: string;
+};
+
+interface RenameProjectFormProps {
+  filename: string;
+  onSubmit: (values: RenameProjectFormValues) => Promise<void>;
+  onCancel: () => void;
+  submitError: string | null;
+}
+
+export default function RenameProjectForm({ filename, onSubmit, onCancel, submitError }: RenameProjectFormProps) {
+  const {
+    handleSubmit,
+    register,
+    formState: { isSubmitting, isDirty, isValid },
+    setFocus,
+  } = useForm<RenameProjectFormValues>({
+    defaultValues: { filename },
+    values: { filename },
+    resetOptions: {
+      keepDirtyValues: true,
+    },
+  });
+
+  useEffect(() => {
+    setFocus('filename');
+  }, []);
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit(onSubmit)} className={style.form}>
+        <FormControl className={style.formControl}>
```

Comment:
question: I dont think the form control does much for us without the associated elements.

We used to use it and decided to move away from it. Cant quite recollect, maybe because it was easier to do it manually with react-hook-form?

I think that, if this doesnt achieve much here, we should aim to remove it

Reply:
fair, let's remove it